ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/MLModelTool.java

```diff
@@ -66,7 +68,13 @@ public MLModelTool(Client client, String modelId, String responseField) {
 
         outputParser = o -> {
             List<ModelTensors> mlModelOutputs = (List<ModelTensors>) o;
-            return mlModelOutputs.get(0).getMlModelTensors().get(0).getDataAsMap().get(responseField);
+            Map<String, ?> dataAsMap = mlModelOutputs.get(0).getMlModelTensors().get(0).getDataAsMap();
+            // Return the response field if it exists, otherwise return the whole response a json string.
+            if (dataAsMap.containsKey(responseField)) {
```

Comment:
Original code doesn't have check on dataAsMap which might cause NPE in edge case, it would be good to add a check here.

Reply:
I see many places that access dataAsMap directly without any index checking, so I follow that rule. But I can add array index checking.